File: src/main/java/net/guizhanss/ultimategenerators2/utils/Keys.java
package net.guizhanss.ultimategenerators2.utils;

import org.bukkit.NamespacedKey;

import net.guizhanss.ultimategenerators2.UltimateGenerators2;

import lombok.experimental.UtilityClass;

@UtilityClass
public final class Keys {
    public static final NamespacedKey ITEMGROUP_MAIN = createKey("main");
    public static final NamespacedKey ITEMGROUP_MISC = createKey("misc");
    public static final NamespacedKey ITEMGROUP_MACHINES = createKey("machines");
    public static final NamespacedKey ITEMGROUP_ELECTRICITY_STORAGE = createKey("electricity_storage");
    public static final NamespacedKey ITEMGROUP_SIMPLE_GENERATORS = createKey("simple_generators");
    public static final NamespacedKey ITEMGROUP_MODULAR_GENERATORS = createKey("modular_generators");
    public static final NamespacedKey CANNOT_CRAFT = createKey("cannot_craft");
    public static final NamespacedKey DIESEL_REFINERY = createKey("diesel_refinery");
    public static final NamespacedKey BIOMASS_EXTRACTION_MACHINE = createKey("biomass_extraction_machine");
    public static final NamespacedKey BIOFUEL_REFINERY = createKey("biofuel_refinery");
    public static final NamespacedKey HEAVY_WATER_REFINING_MACHINE = createKey("heavy_water_refining_machine");

    public static final String ENHANCED_ENDER_CRYSTAL = "UG2_ENHANCED_ENDER_CRYSTAL";

    private static NamespacedKey createKey(String key) {
        return new NamespacedKey(UltimateGenerators2.getInstance(), key);
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/utils/consts/Keys.kt
package net.guizhanss.ultimategenerators2.utils.consts

import net.guizhanss.ultimategenerators2.UltimateGenerators2
import org.bukkit.NamespacedKey

object Keys {
    val ITEMGROUP_MAIN = "main".createKey()
    val ITEMGROUP_MISC = "misc".createKey()
    val ITEMGROUP_MACHINES = "machines".createKey()
    val ITEMGROUP_ELECTRICITY_STORAGE = "electricity_storage".createKey()
    val ITEMGROUP_SIMPLE_GENERATORS = "simple_generators".createKey()
    val ITEMGROUP_MODULAR_GENERATORS = "modular_generators".createKey()
    val CANNOT_CRAFT = "cannot_craft".createKey()
    val DIESEL_REFINERY = "diesel_refinery".createKey()
    val BIOMASS_EXTRACTION_MACHINE = "biomass_extraction_machine".createKey()
    val BIOFUEL_REFINERY = "biofuel_refinery".createKey()
    val HEAVY_WATER_REFINING_MACHINE = "heavy_water_refining_machine".createKey()

    const val ENHANCED_ENDER_CRYSTAL = "UG2_ENHANCED_ENDER_CRYSTAL"

    private fun String.createKey() = NamespacedKey(UltimateGenerators2.getInstance(), this)
}

